app/controllers/peptidome_controller.rb

```diff
@@ -47,6 +47,20 @@ def get_sequences
     render json: Oj.dump(Sequence.list_sequences(ids).join("\n"), mode: :compat)
   end
 
+  # Returns a list of proteins
+  def get_proteins
+    ids = ProteomeCache.delta_decode(JSON(params[:sequence_ids]))
+    sequences = Sequence.includes(original_peptides: :uniprot_entry).where(id: ids)
+    data = Hash[sequences.map { |sequence| [sequence, sequence.original_peptides.map(&:uniprot_entry)] }]
```

Comment:
You are right. I did this because of the easy key/value iteration using `data.each` but forgot about the automatic destructuring in Ruby. 


_[Original comment](https://github.ugent.be/unipept/unipept/pull/576#discussion_r4516) by @bmesuere on Tue Jun 07 2016 at 10:40._